prometheus_es_exporter/__init__.py

```diff
@@ -402,6 +402,56 @@ def conv(value):
     'enum': configparser_enum_conv(('preserve', 'drop', 'zero'))
 }
 
+def create_es_client(es_cluster, ca_certs, client_cert, client_key, headers,
+                     headers_failover, http_auth, http_auth_failover, verify_certs=True):
+    """
+    Create an Elasticsearch client with primary and failover credentials.
+    """
+    def instantiate_client(auth, current_headers):
+        if ca_certs:
+            return Elasticsearch(
+                es_cluster,
+                verify_certs=True,
+                ca_certs=ca_certs,
+                client_cert=client_cert,
+                client_key=client_key,
+                headers=current_headers,
+                http_auth=auth
+            )
+        else:
+            return Elasticsearch(
+                es_cluster,
+                verify_certs=verify_certs,
+                headers=current_headers,
+                http_auth=auth
+            )
+
+    def check_cluster_health(client):
+        try:
+            # Checking the _cluster/health endpoint (GET request)
+            health = client.cluster.health(request_timeout=10)
+            return True
+        except Exception as e:
+            log.error("Cluster health check failed: %s", e)
+            return False
+
+    es_client = instantiate_client(http_auth, headers)
+    try:
+        if not check_cluster_health(es_client):
+            raise Exception("Cluster health check failed for auth credentials.")
+    except Exception as e:
+        log.error("Authentication failed: %s", e)
+        if http_auth_failover:
+            log.error("Falling back to failover auth credentials.")
+            es_client = instantiate_client(http_auth_failover, headers_failover)
+            if not check_cluster_health(es_client):
+                raise Exception("Cluster health check failed for failover credentials as well. Abort!")
+        else:
+            # No secondary credentials provided
+            raise
```

Comment:
This seems a bit off to me. Why raise an exception if you are already checking the boolean value from the `check_cluster_health` func? It returns OK on successful check, otherwise false. So if it is false we should log "auth failed" right and attempt failover?
```suggestion
    if not check_cluster_health(es_client):
        log.error("Authentication failed: %s", e)
        if http_auth_failover:
            log.error("Falling back to failover auth credentials.")
            es_client = instantiate_client(http_auth_failover, headers_failover)
            if not check_cluster_health(es_client):
                raise Exception("Cluster health check failed for failover credentials as well. Abort!")
        else:
            # No secondary credentials provided
            raise
```

Reply:
Good point! I used before a `.ping()` method and no dedicated function to test the credentials, so the try/except block is not required anymore. Changed!